scipy/optimize/_qap.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+from . import linear_sum_assignment, OptimizeResult
+
+
+def umeyama_eigendecomposition(A, B, maximize=False):
+    """Finds an approximate solution to QAP using Umeyama's method."""
+    # 'Correct' non-hermitian matrices
+    if (A != A.T).any() or (B != B.T).any():
+        A = (A + A.T) / 2 + 1j * (A - A.T) / 2
+        B = (B + B.T) / 2 + 1j * (B - B.T) / 2
+
+    va = np.linalg.eigh(A)[1]
+    vb = np.linalg.eigh(B)[1]
+    W = np.abs(vb) @ np.abs(va).T
+    _, col_ind = linear_sum_assignment(W, maximize=maximize)
+
```

Comment:
```suggestion
   col_ind = np.argsort(col_ind)
```
Umeyama's Hungarian algorithm looks for solutions to:
![image](https://user-images.githubusercontent.com/6570539/89694856-da8cb480-d8c6-11ea-8f8a-e6fc71965734.png)

Wherease `linear_sum_assignment`:
![image](https://user-images.githubusercontent.com/6570539/89694875-eb3d2a80-d8c6-11ea-8a9c-794f8c115d68.png)

Consequently our permutation matrix is the transpose of Umeyama's.